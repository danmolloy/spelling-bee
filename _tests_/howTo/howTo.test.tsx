import "@testing-library/jest-dom"
import { screen, render, act, fireEvent } from "@testing-library/react"
import HowTo, { HowToProps } from "../../components/howTo"

const mockProps: HowToProps = {
  setShowMenuItem: jest.fn()
}

describe("", () => {
  beforeEach(() => {
    render(<HowTo {...mockProps} />)
  })
  it("how-to-play is in the document", () => {
    const howToPlay = screen.getByTestId("how-to-play")
    expect(howToPlay).toBeInTheDocument()
  })
  it("menu-icon is in the document and calls setShowMenuItem on click", () => {
    const menuIcon = screen.getByTestId("close-btn")
    expect(menuIcon).toBeInTheDocument()
    act(() => {
      fireEvent.click(menuIcon)
    })
    expect(mockProps.setShowMenuItem).toHaveBeenCalledWith(null)
  })
  it("rules-section is in the document", () => {
    const rulesSection = screen.getByTestId("rules-section")
    expect(rulesSection).toBeInTheDocument()
    expect(rulesSection.textContent).toMatch("Rules")
  })
  it("scoring-section is in the document", () => {
    const scoringSection = screen.getByTestId("scoring-section")
    expect(scoringSection).toBeInTheDocument()
    expect(scoringSection.textContent).toMatch("Scoring")
  })
  it("matches snapshot", () => {
    const howToPlay = screen.getByTestId("how-to-play")
    expect(howToPlay).toMatchSnapshot()
  })
})