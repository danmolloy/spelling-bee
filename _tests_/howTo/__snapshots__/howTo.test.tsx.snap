// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` matches snapshot 1`] = `
<div
  class="h-screen z-40 shadow w-full backdrop-blur-sm absolute flex flex-col items-center"
  data-testid="how-to-play"
>
  <div
    class="rounded mx-2 mt-24 p-4 pb-8 md:mx-24 lg:mx-60 shadow-md border flex flex-col  bg-white text-zinc-700 dark:bg-zinc-900 dark:text-zinc-300 dark:border dark:border-zinc-800"
  >
    <div
      class="flex flex-row w-full justify-between items-center"
    >
      <h1
        class="text-2xl font-bold "
      >
        How to Play 
      </h1>
      <button
        class="hover:text-yellow-500 text-xl m-2 w-10 h-10 rounded-full flex items-center justify-center"
        data-testid="close-btn"
      >
        <svg
          fill="currentColor"
          fill-rule="evenodd"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M799.855 166.312c.023.007.043.018.084.059l57.69 57.69c.041.041.052.06.059.084a.118.118 0 0 1 0 .069c-.007.023-.018.042-.059.083L569.926 512l287.703 287.703c.041.04.052.06.059.083a.118.118 0 0 1 0 .07c-.007.022-.018.042-.059.083l-57.69 57.69c-.041.041-.06.052-.084.059a.118.118 0 0 1-.069 0c-.023-.007-.042-.018-.083-.059L512 569.926 224.297 857.629c-.04.041-.06.052-.083.059a.118.118 0 0 1-.07 0c-.022-.007-.042-.018-.083-.059l-57.69-57.69c-.041-.041-.052-.06-.059-.084a.118.118 0 0 1 0-.069c.007-.023.018-.042.059-.083L454.073 512 166.371 224.297c-.041-.04-.052-.06-.059-.083a.118.118 0 0 1 0-.07c.007-.022.018-.042.059-.083l57.69-57.69c.041-.041.06-.052.084-.059a.118.118 0 0 1 .069 0c.023.007.042.018.083.059L512 454.073l287.703-287.702c.04-.041.06-.052.083-.059a.118.118 0 0 1 .07 0Z"
          />
        </svg>
      </button>
    </div>
    <p
      class="text-lg  "
    >
      Create words using letters from the hive.
    </p>
    <div
      data-testid="rules-section"
    >
      <h2
        class="text-xl font-semibold"
      >
        Rules
      </h2>
      <div
        class="px-2 pb-2"
      >
        <p>
          • Words must use the center letter and be at least 4 letters long.
        </p>
        <p>
          • You can use letters more than once (e.g. 
          <i>
            aardvark
          </i>
          ).
        </p>
      </div>
    </div>
    <div
      data-testid="scoring-section"
    >
      <h2
        class="text-xl font-semibold"
      >
        Scoring
      </h2>
      <div
        class="px-2"
      >
        <p>
          4-letter words are worth 1 point each.
        </p>
        <p>
          Longer words earn 1 point per letter.
        </p>
        <p>
          Each puzzle contains at least one pangram, which is a word using each letter at least once. These are worth an additional 7 points.
        </p>
      </div>
      <div
        class="px-2 pb-2 mt-4"
        data-testid="misc-section"
      >
        <p>
          Found words are saved allowing you to return to the game throughout the day.
        </p>
        <p>
          The game is reset with new letters each day at 8:00am GMT.
        </p>
      </div>
    </div>
  </div>
</div>
`;
